apps/web/src/components/CookieBanner/CookieBanner.tsx

```diff
@@ -0,0 +1,163 @@
+    setCookie(SE_GDPR_COOKIE_NAME, SE_GDPR_COOKIE_ACCEPT_VALUE, { expires: getGDPRCookieExpiryDate(), secure: true })
+    window.gtag('consent', 'update', {
+      ad_storage: 'granted',
+      analytics_storage: 'granted',
+      ad_user_data: 'granted',
+      ad_personalization: 'granted',
+    })
+    document.dispatchEvent(new CustomEvent(COOKIE_SETTINGS_CHANGE_EVENT, { detail: 1 }))
+  }
+
+  const handleReject: MouseEventHandler = () => {
+    setView(CookieBannerView.Rejected)
+    setCookie(SE_GDPR_COOKIE_NAME, SE_GDPR_COOKIE_REJECT_VALUE, { expires: getGDPRCookieExpiryDate(), secure: true })
+    window.gtag('consent', 'update', {
+      ad_storage: 'denied',
+      analytics_storage: 'denied',
+      ad_user_data: 'denied',
+      ad_personalization: 'denied',
+    })
+    document.dispatchEvent(new CustomEvent(COOKIE_SETTINGS_CHANGE_EVENT, { detail: 0 }))
+  }
+
+  const handleHide: MouseEventHandler = () => {
+    setView(CookieBannerView.Hidden)
+  }
+
+  const renderSelection = () => {
+    return (
+      <div className="govuk-cookie-banner__message govuk-width-container">
+        <div className="govuk-grid-row">
+          <div className="govuk-grid-column-two-thirds">
+            <h2 className="govuk-cookie-banner__heading govuk-heading-m">Cookies on Sponsor Engagement</h2>
+            <div className="govuk-cookie-banner__content govuk-body focus:outline-0" ref={regionRef} tabIndex={-1}>
+              <p>We may use some essential cookies to make this service work.</p>
+              <p>
+                We’d like to use additional cookies so we can remember your settings, understand how people use the
+                Sponsor Engagement Tool and make improvements.
+              </p>
+            </div>
+          </div>
+        </div>
+
+        <div className="govuk-button-group mb-0">
+          <button className="govuk-button" name="cookies" onClick={handleAccept} type="button" value="accept">
+            Accept additional cookies
+          </button>
+          <button className="govuk-button" name="cookies" onClick={handleReject} type="button" value="reject">
+            Reject additional cookies
+          </button>
+          <Link className="govuk-link" href="#cookie-policy">
+            View cookie policy
+          </Link>
+        </div>
+      </div>
+    )
+  }
+
+  const renderConfirmation = () => {
+    return (
+      <div className="govuk-cookie-banner__message govuk-width-container">
+        <div className="govuk-grid-row">
+          <div className="govuk-grid-column-two-thirds">
+            <div className="govuk-cookie-banner__content focus:outline-0" ref={regionRef} tabIndex={-1}>
+              <p className="govuk-body" data-testid="confirmation-message">
+                You’ve {view === CookieBannerView.Accepted ? 'accepted' : 'rejected'} additional cookies. You can view
+                the{' '}
+                <Link className="govuk-link" href="#cookie-policy">
```

Comment:
where is this link pointing too?  There is no id of `cookie-policy`

Reply:
the page the link is to go to has not yet been created, the current href is a placeholder, used id so that it would still be shown on url without causing issues

Reply:
When is it likely to be available?
Can we wait until we have the correct policy URL before merging?

The URL should be a configuration item anyway as it may change in future / per environment.

Can you make that change please?

Reply:
I have changed the cookie policy link to what was provided.
I have not changed it to a configuration item, I changed it to a constant to be consistent with the way the app is written.
[https://github.com/PA-NIHR-CRN/sponsor-engagement-web/blob/main/apps/web/src/constants/routes.ts]( constants page )